test/new-e2e/tests/windows/install-test/upgrade_test.go

```diff
@@ -73,6 +75,79 @@ func (s *testUpgradeSuite) TestUpgrade() {
 	s.uninstallAgentAndRunUninstallTests(t)
 }
 
+// TestUpgrade tests upgrading the agent from WINDOWS_AGENT_VERSION to UPGRADE_TEST_VERSION
+func TestUpgradeFromLatest(t *testing.T) {
+	s := &testUpgradeFromLatestSuite{}
+	upgradeAgentPackge, err := windowsAgent.GetUpgradeTestPackageFromEnv()
+	require.NoError(t, err, "should get last stable agent package from env")
+	s.upgradeAgentPackge = upgradeAgentPackge
+	run(t, s)
+}
+
+type testUpgradeFromLatestSuite struct {
+	baseAgentMSISuite
+	upgradeAgentPackge *windowsAgent.Package
+}
+
+func (s *testUpgradeFromLatestSuite) TestUpgradeFromLatest() {
+	vm := s.Env().RemoteHost
+
+	// install current version
+	if !s.Run(fmt.Sprintf("install %s", s.AgentPackage.AgentVersion()), func() {
+		_, err := s.InstallAgent(vm,
+			windowsAgent.WithPackage(s.AgentPackage),
+			windowsAgent.WithInstallLogFile(filepath.Join(s.SessionOutputDir(), "install.log")),
+			windowsAgent.WithValidAPIKey(),
+		)
+		s.Require().NoError(err, "Agent should be %s", s.AgentPackage.AgentVersion())
+	}) {
+		s.T().FailNow()
+	}
+
+	// upgrade to test agent
+	if !s.Run(fmt.Sprintf("upgrade to %s", s.upgradeAgentPackge.AgentVersion()), func() {
+		_, err := s.InstallAgent(vm,
+			windowsAgent.WithPackage(s.upgradeAgentPackge),
+			windowsAgent.WithInstallLogFile(filepath.Join(s.SessionOutputDir(), "upgrade.log")),
+			windowsAgent.WithValidAPIKey(),
+		)
+		s.Require().NoError(err, "should upgrade to agent %s", s.upgradeAgentPackge.AgentVersion())
+	}) {
+		s.T().FailNow()
+	}
+
+	// run tests
+	testerOptions := []TesterOption{
+		WithAgentPackage(s.upgradeAgentPackge),
+	}
+	t, err := NewTester(s, vm, testerOptions...)
+	s.Require().NoError(err, "should create tester")
+	if !t.TestInstallExpectations(s.T()) {
+		s.T().FailNow()
+	}
+
+	// Get Display Version
+	productVersion, err := windowsAgent.GetDatadogProductVersion(vm)
+	s.Require().NoError(err, "should get product version")
+
+	// split version string by .
+	versionParts := strings.Split(productVersion, ".")
+
+	// verify there are four parts
+	require.Len(s.T(), versionParts, 4, "Version should have four parts")
+
+	// get version parts of the upgrade test version
+	ver, _ := version.New(strings.TrimSuffix(s.upgradeAgentPackge.Version, "-1"), "")
+
+	// check if the version is the same as the upgrade test version
+	patchVersion, err := strconv.Atoi(versionParts[2])
+	s.Require().NoError(err, "should convert patch version to int")
+	assert.Equal(s.T(), ver.Patch+1, int64(patchVersion), "Patch version should increment by 1")
```

Comment:
suggestion: instead of comparing to the package version, which doesn't seem to be correct here since it also borrows `WINDOWS_AGENT_VESION`, you could call `windowsAgent.GetDatadogProductVersion(vm)` once after the initial install and then again after the upgrade, and compare those two versions.